File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/commands/UpdateTrustlyAccountCommand.java
package com.hedvig.paymentservice.domain.payments.commands;

import lombok.Value;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

import java.util.UUID;

@Value
public class UpdateTrustlyAccountCommand {
    @TargetAggregateIdentifier
    String memberId;

    UUID hedvigOrderId;

    String accountId;
    String address;
    String bank;
    String city;
    String clearingHouse;
    String descriptor;
    boolean directDebitMandateActive;
    String lastDigits;
    String name;
    String personId;
    String zipCode;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/commands/UpdateTrustlyAccountCommand.kt
package com.hedvig.paymentservice.domain.payments.commands

import java.util.*
import lombok.Value
import org.axonframework.commandhandling.TargetAggregateIdentifier

@Value
class UpdateTrustlyAccountCommand(
    @TargetAggregateIdentifier  
    val memberId: String,

    val hedvigOrderId: UUID,

    val accountId: String,
    val address: String?,
    val bank: String?,
    val city: String?,
    val clearingHouse: String?,
    val descriptor: String?,
    val directDebitMandateActive: Boolean?,
    val lastDigits: String?,
    val name: String?,
    val personId: String?,
    val zipCode: String?
)
